From david-b@pacbell.net Tue Jun 13 21:00:58 2006
Date: Tue, 13 Jun 2006 20:55:44 -0700
From: Milan Svoboda <msvoboda@ra.rockwell.com>
Message-Id: <200606132055.45525.david-b@pacbell.net>
Subject: usb gadget: allow drivers support speeds higher than full speed
Date: Monday 29 May 2006 3:34 am
To: linux-usb-devel@lists.sourceforge.net

From: Milan Svoboda <msvoboda@ra.rockwell.com>

This patch removes limitation which prevents use of drivers that support
speeds different that full speed.


Signed-off-by: Milan Svoboda <msvoboda@ra.rockwell.com>
Signed-off-by: David Brownell <dbrownell@users.sourceforge.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/usb/gadget/pxa2xx_udc.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/usb/gadget/pxa2xx_udc.c
+++ gregkh-2.6/drivers/usb/gadget/pxa2xx_udc.c
@@ -1585,7 +1585,7 @@ int usb_gadget_register_driver(struct us
 	int			retval;
 
 	if (!driver
-			|| driver->speed != USB_SPEED_FULL
+			|| driver->speed < USB_SPEED_FULL
 			|| !driver->bind
 			|| !driver->unbind
 			|| !driver->disconnect
